simple/lint.go

```diff
@@ -1807,3 +1807,21 @@ func LintSimplifyTypeSwitch(pass *analysis.Pass) (interface{}, error) {
 	pass.ResultOf[inspect.Analyzer].(*inspector.Inspector).Preorder([]ast.Node{(*ast.TypeSwitchStmt)(nil)}, fn)
 	return nil, nil
 }
+
+func LintRedundantCanonicalHeaderKey(pass *analysis.Pass) (interface{}, error) {
+	fn := func(node ast.Node) {
+		call := node.(*ast.CallExpr)
+		for _, function := range []string{"Add", "Del", "Get", "Set"} {
+			fqdn := fmt.Sprintf("(net/http.Header).%s", function)
```

Comment:
Instead of looping over a slice and constructing the fqdn over and over, let's just write the `if` by hand:

```
if !IsCallToAST(pass, call, "(net/http.Header).Add") &&
    !IsCallToAST(pass, call, "(net/http.Header).Del") &&
    ...
```

This has the benefit of making it clear that only up to one element of the slice can ever match. That is, if we were to keep using the loop, we'd want to return after ReportNodeFG, and we'd want to return, instead of continue, when the first argument isn't a call to CanonicalHeaderKey. Using the expanded boolean expression avoids all that.